docs/csharp/codedoc.md

```diff
@@ -2,26 +2,1348 @@
+The compiler generated XML file can be distributed alongside your .NET assembly so that Visual Studio and other IDEs can show quick information about types or members when performing intellisense.
+Additionally the XML file can be run through tools like [DocFX](https://dotnet.github.io/docfx/) and [Sandcastle](https://github.com/EWSoftware/SHFB) to generate full on API reference websites.
+
+XML documentation comments like all other comments are ignored by the compiler, to enable generation of the XML file add `"xmlDoc":true` under `buildOptions` in your `project.json` when using .NET Core or use the `/doc` compiler option for the .NET framework.
+Go [here](https://msdn.microsoft.com/en-us/library/3260k4x7.aspx) to learn how to enable XML documentation generation in Visual Studio.
+
+XML documentation comments are characterized by triple forward slashes (`///`) and an XML formatted comment body.
+
+```csharp
+/// <summary>
+/// This class does something.
+/// </summary>
+public class SomeClass
+{
+
+}
+```
+
+## Walkthrough
+
+In this section I'm going to walk you through documenting a very basic math library to make it easy for new developers to understand/contribute and for third party developers to use.
+We're going to use a combination of single line comments, multiline comments and XML documentation tags.
+
+Here's code for the simple math library:
+
+```csharp
+public class Math
+{
+    public static int Add(int a, int b)
+    {
+        if ((a == int.MaxValue && b > 0) || (b == int.MaxValue && a > 0))
+            throw new System.OverflowException();
+
+        return a + b;
+    }
+
+    public static double Add(double a, double b)
+    {
+        if ((a == double.MaxValue && b > 0) || (b == double.MaxValue && a > 0))
+            throw new System.OverflowException();
+
+        return a + b;
+    }
+
+    public static int Subtract(int a, int b)
+    {
+        return a - b;
+    }
+
+    public static double Subtract(double a, double b)
+    {
+        return a - b;
+    }
+
+    public static int Multiply(int a, int b)
+    {
+        return a * b;
+    }
+
+    public static double Multiply(double a, double b)
+    {
+        return a * b;
+    }
+
+    public static int Divide(int a, int b)
+    {
+        return a / b;
+    }
+
+    public static double Divide(double a, double b)
+    {
+        return a / b;
+    }
+}
+```
+
+The sample library above doesn't do much, it supports four major arithmetic operations `add`, `subtract`, `multiply` and `divide` on `int` and `double` datatypes.
+Now let's assume this is one huge library that is going to have a ton of developers, we'll need to add some comments to make it easier to understand.
+In this sample we'll use multiline comments for class definitions and single line comments otherwise, these are not imposed conventions however,
+so feel free to use the kind of comment the situation requires.
+
+Here's the updated library code with comments added:
```

Comment:
I'd start with this version. The focus here is on XML documentation. Explaining the syntax for comments is a bit of a distraction. The discussion from here on is the key information in the article.